integrations/launchdarkly/client.py

```diff
@@ -0,0 +1,218 @@
+
+PAGE_SIZE = 100
+
+
+class ObjectKind(StrEnum):
+    PROJECT = "project"
+    AUDITLOG = "auditlog"
+    FEATURE_FLAG = "flag"
+    ENVIRONMENT = "environment"
+
+
+class LaunchDarklyClient:
+    def __init__(
+        self, api_token: str, launchdarkly_url: str = "https://app.launchdarkly.com"
+    ):
+        self.api_url = f"{launchdarkly_url}/api/v2"
+        self.api_token = api_token
+        self.http_client = http_async_client
+        self.http_client.headers.update(self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {
+            "Authorization": f"{self.api_token}",
+            "Content-Type": "application/json",
+        }
+
+    async def get_paginated_resource(
+        self, kind: str, resource_path: str | None = None, limit: int = PAGE_SIZE
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        kind = kind + "s"
+        url = kind if not resource_path else f"{kind}/{resource_path}"
+        url = url.replace("auditlogs", "auditlog")
+
+        params = {"limit": limit}
+
+        while url:
+            try:
+                response = await self.send_api_request(
+                    endpoint=url, query_params=params
+                )
+                items = response.get("items", [])
+                yield items
+
+                if "_links" in response and "next" in response["_links"]:
+                    url = response["_links"]["next"]["href"]
+                else:
+                    total_count = response.get("totalCount")
+                    logger.info(f"Fetched {total_count} {kind} from Launchdarkly")
+                    break
+
+            except httpx.HTTPStatusError as e:
+                logger.error(
+                    f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+                )
+                raise
+            except httpx.HTTPError as e:
+                logger.error(
+                    f"HTTP error occurred while fetching {kind} from LaunchDarkly: {e}"
+                )
+                raise
+
+    async def send_api_request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        logger.info(f"Requesting Launchdarkly data for endpoint: {endpoint}")
+        try:
+            endpoint = endpoint.replace("/api/v2/", "")
+            url = f"{self.api_url}/{endpoint}"
+            logger.info(
+                f"URL: {url}, Method: {method}, Params: {query_params}, Body: {json_data}"
+            )
+            response = await self.http_client.request(
+                method=method,
+                url=url,
+                params=query_params,
+                json=json_data,
+            )
+            response.raise_for_status()
+
+            logger.info(f"Successfully retrieved data for endpoint: {endpoint}")

```

Comment:
debug

Reply:
Mind telling why its debug and not info ?